libs/payments/stripe/src/lib/stripe.util.ts

```diff
@@ -40,7 +36,9 @@ export const checkSubscriptionPromotionCodes = (
     );
   }
   if (!validPromotionCodes.includes(code)) {
-    throw new PromotionCodeNotForSubscriptionError();
+    throw new PromotionCodeCouldNotBeAttachedError(
+      "Promotion code restricted to a product that doesn't match the product on this subscription"
```

Comment:
What promotion code? What product? Adding either/both of these to the error will make the Sentry error easier to trouble-shoot.